### PR TITLE
chore(packages): Adding checkVerifiedContact and handleResendConfirmationCode to defaultServices

### PR DESCRIPTION
**Pull request by avirajdh**
#### Description of changes
This PR adds two new functions(`checkVerifiedContact` and `handleResendConfirmationCode`) to `defaultServices`. It is related to this [PR](https://github.com/aws-amplify/amplify-ui/pull/2546).

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/data/amplifyMock/components.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
+import { Flex, Icon, Button, Text, Divider } from '@aws-amplify/ui-react';
```
Comment (wlee221) [inline docs/src/data/amplifyMock/components.tsx:1]: Didn't we already add this?

Comment (wlee221) [reply docs/src/data/amplifyMock/components.tsx:1]: Maybe sync your remote branch (`avirajdh:poc-authenticator-demo`) with upstream (`aws-amplify:poc-authenticator-demo`)

File: packages/ui/src/machines/authenticator/defaultServices.ts [hunk 2]
```diff
@@ -49,6 +54,14 @@ export const defaultServices = {
   }): Promise<any> {
     return await Auth.confirmSignUp(username, code);
   },
+  /** @deprecated For internal use only */
+  async handleResendConfirmationCode({
+    username,
+  }: {
+    username: string;
+  }): Promise<any> {
```
Comment (wlee221) [inline packages/ui/src/machines/authenticator/defaultServices.ts:62]: Can we remove `Promise<any>` here?

```suggestion
  }) {
```

Idea is that TS will automatically infer the type for us.

File: packages/ui/src/machines/authenticator/signUp.ts [hunk 3]
```diff
@@ -257,8 +257,8 @@ export function createSignUpMachine({ services }: SignUpMachineOptions) {
           const { user, authAttributes } = context;
           const username =
             get(user, 'username') || get(authAttributes, 'username');
-
-          return Auth.resendSignUp(username);
+          const result = await services.handleResendConfirmationCode(username);
+          return result;
```
Comment (wlee221) [inline packages/ui/src/machines/authenticator/signUp.ts:261]: Nit: 
```suggestion
          return services.handleResendConfirmationCode(username);
```

File: packages/ui/src/machines/authenticator/actors/signIn.ts [hunk 4]
```diff
@@ -582,8 +582,7 @@ export function signInActor({ services }: SignInMachineOptions) {
         },
         async checkVerifiedContact(context, event) {
           const { user } = context;
-          const result = await Auth.verifiedContact(user);
-
+          const result = await services.checkVerifiedContact(user);
           return result;
```
Comment (wlee221) [inline packages/ui/src/machines/authenticator/actors/signIn.ts:586]: Nit: return `services.checkVerifiedContact(user)` directly

File: packages/ui/src/machines/authenticator/defaultServices.ts [hunk 5]
```diff
@@ -40,6 +40,11 @@ export const defaultServices = {
   }): Promise<any> {
     return Auth.confirmSignIn(user, code, mfaType);
   },
+  /** @deprecated For internal use only */
```
Comment (zchenwei) [inline packages/ui/src/machines/authenticator/defaultServices.ts:43]: We should use `@internal`

File: packages/ui/src/machines/authenticator/defaultServices.ts [hunk 6]
```diff
@@ -49,6 +54,10 @@ export const defaultServices = {
   }): Promise<any> {
     return await Auth.confirmSignUp(username, code);
   },
+  /** @deprecated For internal use only */
```
Comment (zchenwei) [inline packages/ui/src/machines/authenticator/defaultServices.ts:57]: Plz use `@internal`

File: packages/ui/src/machines/authenticator/defaultServices.ts [hunk 7]
```diff
@@ -49,6 +54,10 @@ export const defaultServices = {
...
+  async handleResendConfirmationCode({ username }: { username: string }) {
```
Comment (zchenwei) [inline packages/ui/src/machines/authenticator/defaultServices.ts:58]: I do not think the argument gets passed in is an object. You should not destructure here

Comment (wlee221) [reply packages/ui/src/machines/authenticator/defaultServices.ts:58]: Oh yeah, the arguments should match the underlying auth call. So +1 with

```suggestion
  async handleResendConfirmationCode(username: string) {
```

**Review by zchenwei — APPROVED** [review]

**Pull request merged**